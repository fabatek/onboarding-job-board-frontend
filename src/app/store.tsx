import { configureStore } from '@reduxjs/toolkit';
import jobReducer from '../features/job/jobSlice';

export default configureStore({
  reducer: {
    job: jobReducer,
  },
});
