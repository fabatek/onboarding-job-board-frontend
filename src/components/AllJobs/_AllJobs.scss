.jobsCard{
    display: flex;
    width: 80%;
    height: 15%;
    margin-left: 10%;
    box-shadow: 10px 10px 20px gray;
    border-radius: 30px;
    margin-top: 30px;
    .jobsCard__left{
       width: 20%;
        img{
            max-width: 100%;
            border-radius: 30px 0 0 30px ;
            height: 100%;
        }
    }
    .jobsCard__right{
        padding-right: auto;
        margin-right: 2%;
        text-align: right;
    }
    .jobsCard__center{
        display: flex;
        flex-direction: column;
        text-align: left;
        margin-left: 5%;
        
        width: 70%;
        .name,.company,.type{
            color: black;
            text-decoration: none;
        }
        .name{
            font-size: 25px;
            font-weight: 500;
            &:hover{
                color: brown;
            }
        }
        .company,.type{
            line-height: 2.5;
        }
    }
    
}