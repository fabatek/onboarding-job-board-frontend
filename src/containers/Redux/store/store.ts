import { configureStore } from "@reduxjs/toolkit";
import jobSlice from "../jobs/jobSlice";
import searchSlice from "../search/searchSlice";
export const store = configureStore({
  reducer: {
    jobs: jobSlice,
    search: searchSlice
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
