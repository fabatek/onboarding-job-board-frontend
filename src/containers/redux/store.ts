import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit'
import { jobSlices } from './jobSlices'
export const store = configureStore({
    reducer: {
        jobs: jobSlices.reducer, 
    },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
    ReturnType,
    RootState,
    unknown,
    Action<string>
>
