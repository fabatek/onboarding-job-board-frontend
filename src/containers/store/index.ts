export * from './store'
export * as jobActions from './actions/action-creators/jobActions';
export * from './reducers'

