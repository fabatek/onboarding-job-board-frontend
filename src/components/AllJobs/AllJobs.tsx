import React, { FC, useEffect, useState } from 'react'
import './_AllJobs.scss'
import axios from 'axios';
import { jobs } from '../../types/jobsType'


const AllJobs: FC = () => {
    const [jobs, setJobs] = useState([]);
    useEffect(() => {
        axios.get('https://6176370c03178d00173daae3.mockapi.io/api/api')
        .then(res => {
            setJobs(res.data)
        }).catch(err => {
            console.log(err);
        })
    },[])
    
    return (
        <div className = "allJobs">
            <h2>All Jobs({jobs.length} Jobs)</h2>
            {jobs.map((jobs:any) => (       
                <div key={jobs.id} className="jobsCard" >
                    <div className="jobsCard__left">
                        <img src={jobs.jobImg} alt="" />
                    </div>
                    <div className = "jobsCard__center">
                        <a href="" id="name" >{jobs.jobName}</a>
                        <a href="" id="company">{jobs.jobCompany}</a>
                        <a href="" id="type">{jobs.jobType}</a>
                    </div>
                    <div className="jobsCard__right">
                        <p>{jobs.jobArea}</p>
                    </div>
                    
                </div>
        ))}
        </div>
        
    )
}

export default AllJobs
