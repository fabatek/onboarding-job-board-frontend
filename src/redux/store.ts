import { applyMiddleware, createStore } from 'redux';
import rootReducers from './root-reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

const store = createStore(
    rootReducers,
    composeWithDevTools(applyMiddleware(thunk)),
);

export type RootStore = ReturnType<typeof rootReducers>


export default store;