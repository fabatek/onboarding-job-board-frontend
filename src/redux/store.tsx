import { applyMiddleware, createStore } from "redux"; 
import thunk from "redux-thunk";
import { jobsReducer } from "./reducer";

export const store = createStore(jobsReducer,applyMiddleware(thunk))