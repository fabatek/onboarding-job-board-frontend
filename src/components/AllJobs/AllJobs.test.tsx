import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import AllJobs from "./AllJobs";
import store from '../../store'

describe("Step form reandering", () => {
  it('it should render a step form`', () => {
    const { container } = render(<Provider store={store}><AllJobs /></Provider>);
    expect(container).toBeInTheDocument();
  });
});

describe("Display jobs test", () => {
  it('renders an `.alljobs`', () => {
    const { container } = render(<Provider store={store}><AllJobs /></Provider>);
    expect(container.querySelectorAll('.alljobs').length).toBe(1);
  });

});

