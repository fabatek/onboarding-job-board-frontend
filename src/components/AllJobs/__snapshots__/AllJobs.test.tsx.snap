// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step form reandering it should render a step form\` 1`] = `
<div>
  <div
    class="allJobs"
  >
    <h2>
      All Jobs(
      0
       Jobs)
    </h2>
  </div>
</div>
`;
